### ISSUE TITLE
1.10.1 Flickers Every 5-10 Seconds

### ISSUE THREAD

**Issue by ghost** [brndnmtthws/conky#170]
Running 1.10.1 I noticed it flickers every few seconds, like maybe every 5-10 seconds.

1.9.0 runs great, ever since the 1.10x versions came out, nothing but trouble for me, I'd really like to figure out what is going on here?

These are some of system specs;

Slackware 14.1 x86_64
Openbox 3.6.1
compton
xorg-server-1.14.3
nvidia-driver-352.41

I'm not sure what would be casuing this, here is what my config looks like;

conky.config = {
override_utf8_locale = false,
background = true,
use_xft = true,
font = 'aller:size=9',
xftalpha = 0.8,
out_to_console = false,
update_interval = 3.0,
total_run_times = 0,
draw_shades = false,

own_window = true,
own_window_class = 'Conky',
own_window_type = 'desktop',    -- normal /override /desktop /dock
own_window_transparent = true,
own_window_hints = 'undecorated,sticky,skip_taskbar,skip_pager,below',

-- Prevent Conky refresh over itslef in tile wms 
-- true - true transparency with compton 
-- false - pseudo-transparency without compton
own_window_argb_visual = true,

minimum_size = 1920,
maximum_width = 1920,

double_buffer = true,
default_color = 'e3412c',
color1 = 'e0cdc3',

alignment = 'top_middle',
gap_y = 5, --vertical
-- gap_x = 0, --horizontal
no_buffers = true
}

conky.text = [[
${alignc}${color}CPU1: ${color1}${cpu cpu0}%   \
${color}CPU2: ${color1}${cpu cpu1}%   \
${color}CPU3: ${color1}${cpu cpu2}%   \
${color}CPU4: ${color1}${cpu cpu3}% - ${freq}MHz - ${hwmon 0 temp 1} °C   \
${color}GPU: ${color1}${execi 100 nvidia-settings -q gpucoretemp -t | head -1} °C   \
${color}RAM: ${color1}${mem} - $memperc%   \
${color}SSD: ${color1}${fs_free /} free   \
${color}LAN: ${color1}${addr eth0} - ${downspeed eth0} kb/s down - ${upspeed eth0} kb/s up   \
${color}POWER: ${color1}${acpiacadapter AC0}  -  ${color}BATTERY: ${color1}${battery_percent BAT0}% ${battery_time BAT0}  ${color}UP: ${color1}$uptime_short \
${execi 70 ~/.conky/scripts/battery_notify.sh}
]]

**Comment by plikhari** [comment]
Try adding ``` text_buffer_size = 512, ``` - the rest of the lines look fine.

**Comment by Geyup** [comment]
I'll give it a try when I have time and report back.

By the way, I forgot, but some of lines I put single quotes around some things. Are these actually needed?

Thanks

**Comment by plikhari** [comment]
Yes the quotes are required - unless they are digits or reserved words like true etc - basically following lua conventions for a lua script/program

**Comment by Geyup** [comment]
Ok can't say I know much about lua conventions...

Well it's still flickering and I changed it to 256, and 1024 and still the same.

Also I noticed at the CLI; ```conky: Unknown setting 'minimum_size'``` so what would I change these two lines to now, or I just use only the max width one?

minimum_size = 1920,
maximum_width = 1920,

thanks

**Comment by Geyup** [comment]
AHHH I changed the config to;

```
minimum_width = 1920,
maximum_width = 1920,
```
And it's not flickering anymore LOL...

Looks like the problem was the ```minimum_size =``` :)

**Comment by plikhari** [comment]
You would change it the new format
``` 	
maximum_width = 1920,
minimum_width = 1920, minimum_height = 1080,
```
The best way to convert your older format to newer 1.10x format of rc files is to use the utility 'convert.lua'. This is located in the conky folder under /usr/share/doc/conky
You can perhaps copy to a folder in your $HOME location - make it executable ``` sudo chmod +x convert.lua``` and then start of and see the new file and compare the changes.

OK appears you solved it - but convert comment and other stuff should help you on your way forward ;-)
cheers

**Comment by Geyup** [comment]
Why ```minimum_height = 1080```? That seems like to fill up the entire screen?

Well this is what my config looks like now, do you think I really need to use convert.lua?

conky.config = {
override_utf8_locale = false,
background = true,
use_xft = true,
font = 'aller:size=9.5',
xftalpha = 0.8,
out_to_console = false,
out_to_stderr = false,
update_interval = 3.0,
total_run_times = 0,
draw_shades = false,

own_window = true,
own_window_class = 'Conky',
own_window_type = 'desktop',
own_window_transparent = true,
own_window_hints = 'undecorated,sticky,skip_taskbar,skip_pager,below',

own_window_argb_visual = true,

minimum_width = 1920,
maximum_width = 1920,

double_buffer = true,
default_color = 'e3412c',
color1 = 'e0cdc3',

alignment = 'top_middle',
gap_y = 5,
no_buffers = true
}

conky.text = [[
${alignc}${color}CPU1: ${color1}${cpu cpu0}%   \
${color}CPU2: ${color1}${cpu cpu1}%   \
${color}CPU3: ${color1}${cpu cpu2}%   \
${color}CPU4: ${color1}${cpu cpu3}% - ${freq}MHz - ${hwmon 0 temp 1} °C   \
${color}GPU: ${color1}${execi 100 nvidia-settings -q gpucoretemp -t | head -1} °C   \
${color}RAM: ${color1}${mem} - $memperc%   \
${color}SSD: ${color1}${fs_free /} free   \
${color}LAN: ${color1}${addr eth0} - ${downspeed eth0} kb/s down - ${upspeed eth0} kb/s up   \
${color}POWER: ${color1}${acpiacadapter AC0}  -  ${color}BATTERY: ${color1}${battery_percent BAT0}% ${battery_time BAT0}  ${color}UP: ${color1}$uptime_short \
${execi 70 ~/.conky/scripts/battery_notify.sh}
]]


thanks

**Comment by Geyup** [comment]
By the way in the doc folder for the conky.conf what does the ```$hr``` stand for?

thanks

**Comment by Geyup** [comment]
I forgot to mention if I run conky from the CLI I get no errors, so my config must be good?

>conky: desktop window (289) is root window
conky: window type - desktop
conky: drawing to created window (0x2400002)
conky: drawing to double buffer
conky: forked to background, pid is 7921

**Comment by Geyup** [comment]
One last thing for now, I noticed in the variables and new config there doesn't seem to be this option anymore...

```total_run_times = ``` is this still being used?

thanks

**Comment by plikhari** [comment]
That 1080 was an example :)
Yes - if you are not getting errors and it is working fine - no worries 
total_run_times is being used and there are some recent issues being reported - I have it set to zero and it is working fine.

**Comment by plikhari** [comment]
$hr is horizontal line

**Comment by Geyup** [comment]
Ok I'll leave the ```total_run_times = 0``` and thanks for every thing, it runs great now!

Cheers

**Comment by Geyup** [comment]
Ok I'll leave the ```total_run_times = 0``` and thanks for every thing, it runs great now!

Cheers

**Comment by plikhari** [comment]
Can you please close this issue as it stands resolved - cheers

**Comment by lasers** [comment]
Resolved. Nice work, mister. Closing.

**Comment by WinEunuuchs2Unix** [comment]
Just installed 1.10 because 1.9 is no longer available in Ubuntu and notice conky screen flickering frequently with skylake processor. Also pre-exec command no longer supported and must resort to exec instead. Although converting 1.9 scripts to 1.10 is an option it should not be necessary for users. That said earlier versions of 1.10 couldn't run my 1.9 conkyrc at all so I had to pin / mark hold an earlier version.

**Comment by lasers** [comment]
>Just installed 1.10 because 1.9 is no longer available in Ubuntu and notice conky screen flickering frequently with skylake processor

If you receive conky screen flickering frequently on 1.10, but not 1.9, then you should try and git bisect this... so you can find the bad commit. You would have to compile `conky` repeatedly between commits.

**Comment by WinEunuuchs2Unix** [comment]
My luck with compiling grub for UEFI left a bad taste in my mouth. Compiling for i386-pc was ok... Anyway changing `maximum_width 400` to `maximum_width 1920` and adding a new line `maximum_height 1080` made the flicker go away. The opaqueness (right word?) got darker though but I can adjust that later. Thanks for quick response :)
maximum_height 1080

**Comment by lasers** [comment]
>Just installed 1.10 because 1.9 is no longer available

`1.10.0`? I would like for you to tell me if this is still happening on `1.10.9_pre` (git). We're using `cmake` now since `1.10.8` or so... Compiling conky should be relatively painless now.

Depending on what variables/object you have in your `conky.conf`, you may have to enable some flags first (eg `cmake -DBUILD_JOURNAL ../`) to use `{journal}`. NVIDIA variables does not work right now.

You could start pointing out some other issues with `conky` that we don't know anything about. :-)

**Comment by lasers** [comment]
I tried the configs posted by **ghost**. No flickering here.
I tried `minimal_width = 100` too to see if it would flicker. Nothing.

![2018-08-13-204258_513x745_scrot](https://user-images.githubusercontent.com/852504/44067442-171a42a4-9f3b-11e8-8403-cdaba59ef6d6.png)

If you're having this problem on `1.10.9` or `1.10.*` (git), comment here with the config. The older versions are not trustworthy due to too many changes that can be hard to track. Thank you. :kissing_heart:
